synapse/events/snapshot.py

```diff
@@ -295,28 +298,30 @@ class _AsyncEventContextImpl(EventContext):
             that was replaced.
     """
 
+    # This needs to have a default as we're inheriting
+    _storage = attr.ib(default=None)
```

Comment:
Annoyingly this needs a default, otherwise attr complains (as we're inheriting from a attr class which have attributes with default values). I'm not sure if there's another way round this beyond removing all the defaults.